GrafCalculator/app/src/main/java/dev/software/bing/Evaluate.kt

```diff
@@ -0,0 +1,159 @@
+package dev.software.bing
+
+
+//    PARSER RULES
+//    expression : plusMinus* EOF
+//    plusMinus: mulDiv ( ( '+' | '-' ) mulDiv )*
+//    mulDiv : factor ( ( '*' | '/' ) factor )*
+//    factor : NUMBER | '(' expression ')'
+
+
+
+fun main() {
+    //val expression: String = "105 - 99 + 3* 5 /4 " //9.75
+    //val expression: String = "105 - 99 + 3* (14 - 2* (19 - 3)) /4 " //-7.5
+    val expression: String = "33*9-5*(6+5*4)/3 " //253.66666666666666
+    //val expression = "11*8-5* (2-3)"//93
+    val eval = Evaluate()
+    val lexemeList = eval.parsing(expression)
+    val utilityLexemes = UtilityLexemes(lexemeList)
+    println(eval.express(utilityLexemes))
+}
+
+enum class Type {
+    MINUS, PLUS, DIVISION, MULTIPLICATION, NUMBER, L_BRACKET, R_BRACKET, EOF
+}
+
+
+data class Lexeme(var type: Type, var value: String){
```

Comment:
по возможности лучше использовать неизменяемые поля, если что объект можно скопировать + у тебя есть стабильное состояние и нет никакой магии что в разный момент времени не понятно что содержится в классе
```suggestion
data class Lexeme(val type: Type, val value: String)
```